common/git-common/src/main/java/org/arquillian/pact/common/git/GitOperations.java

```diff
@@ -132,6 +147,65 @@ public Ref checkoutBranch(Git git, String branch, String remote) {
     }
 
     /**
+     * Executes a checkout -b command using given branch.
+     * @param git instance.
+     * @param branch to create and checkout.
+     * @return Ref to current branch.
+     */
+    public Ref createBranchAndCheckout(Git git, String branch)  {
+        try {
+            return git.checkout()
+                    .setCreateBranch(true)
+                    .setName(branch)
+                    .setUpstreamMode(CreateBranchCommand.SetupUpstreamMode.TRACK)
+                    .call();
+        } catch (GitAPIException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    /**
+     * Add all files and commit them with given message. This is equivalent as doing git add . git commit -m "message".
+     * @param git instance.
+     * @param message of the commit.
+     * @return RevCommit of this commit.
+     */
+    public RevCommit addAndCommit(Git git, String message) {
+        try {
+            git.add()
+                    .addFilepattern(".")
+                    .call();
+            return git.commit()
+                    .setMessage(message)
+                    .call();
+        } catch (GitAPIException e) {
+            throw new IllegalArgumentException(e);
```

Comment:
yes, GITException is the superclass of 35 exception classes. So it is not easy to filter them. Maybe we can change to `IllegalStateException`. But since what we want is to have something simple, I would avoid starting doing multiple catch.